File: sol/solution/0301-0400/0303/Solution.rb
class NumArray
=begin
    :type nums: Integer[]
=end
    def initialize(nums)
        @total_sum = 0
        @prefix_sum = []
        nums.each_with_index do |element, index|
            @total_sum += element
            @prefix_sum[index] = @total_sum
        end
    end
=begin
    :type left: Integer
    :type right: Integer
    :rtype: Integer
=end
    def sum_range(left, right)
        left.zero? ? @prefix_sum[right] : @prefix_sum[right] - @prefix_sum[left -1]
    end
end
# Your NumArray object will be instantiated and called as such:
# obj = NumArray.new(nums)
# param_1 = obj.sum_range(left, right)

File: sol/solution/0301-0400/0303/Solution.dart
class NumArray {
  List<int> arr = [];
  NumArray(List<int> nums) {
    List<int> prefixSum = [];
    int currSum = 0;
    for (int i = 0; i < nums.length; i++) {
      currSum += nums[i];
      prefixSum.add(currSum);
    }
    arr = prefixSum;
  }
  int sumRange(int left, int right) {
    if (left - 1 >= 0) {
      return arr[right] - arr[left - 1];
    } else {
      return arr[right];
    }
  }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = NumArray(nums);
 * int param1 = obj.sumRange(left,right);
 */